python-job-crawler.py

```diff
@@ -1,88 +1,143 @@
-def main():
-    print("🏁 Starting the web crawler...")
+logging.basicConfig(  # logger initialization
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s'
+)
+
+SRC_KW_FILENAME = "keywords.txt"  # file containing keywords to look for
+PAGE_LOADING_TIME_SLEEPER = 5  # time in s waiting for the web page to load
+
+def main() -> None:
+    """
+    Main function to initialize the web crawler, configure the Chrome driver,
+    and crawl career pages.
+
+    Raises:
+        FileNotFoundError: If file `SRC_KW_FILENAME` is not found.
+        IOError: If there is an issue opening or reading the file.
+        Exception: Any other exceptions that may occur during file processing.
+    """
+
+    file_path = Path(SRC_KW_FILENAME)
+    keywords = []
+
+    # If the kw file exists, insert its kws inside our keywords list
+    if file_path.exists():
+        try:
+            with file_path.open('r', encoding='utf-8') as f:
+                keywords = [keyword.strip().lower() for keyword in f if keyword.strip()]
+        except (IOError, FileNotFoundError) as e:
+            logging.error(f"Error reading {SRC_KW_FILENAME}: {e}")
+    if not keywords:
+        logging.error("No keywords to search for. Please check your keywords file.")
+        return
+
+    logging.info("🏁 Starting the web crawler...")
 
     # Setup Chrome options
     chrome_options = Options()
-    chrome_options.add_argument('--headless')
-    chrome_options.add_argument('--no-sandbox')
-    chrome_options.add_argument('--disable-dev-shm-usage')
-
-    # Initialize WebDriver with the exact path to chromedriver
-    # Add your specific path to a filed called chromeDriverPath.py.
-    # The contents of chromeDriverPath.py should look similar to this:
-    # chromeDriverPath = '/Users/sueellenmisky/.wdm/drivers/chromedriver/mac64/127.0.6533.119/chromedriver-mac-x64/chromedriver'
-    # Ensure chromeDriverPath.py is in your .gitignore
+    options_list = ['--headless', '--no-sandbox', '--disable-dev-shm-usage']
+    for opt in options_list:
+        chrome_options.add_argument(opt)
+
+    # Initialize WebDriver using ChromeDriverManager
     service = Service(chromeDriverPath)
     driver = webdriver.Chrome(service=service, options=chrome_options)
 
-    keywords_found_count = 0
+    # Get the number of match
+    kws_found_count = 0
     for careerPage in careerPages:
-        if crawl_careerPage(careerPage, driver):
-            keywords_found_count += 1
-        # Add a delay between requests
-        time.sleep(1)  
+        if crawl_careerPage(careerPage, driver, keywords):
+            kws_found_count += 1
+        time.sleep(1)
 
     driver.quit()
-    print(f"Keywords were found in job listings at {keywords_found_count} out of {len(careerPages)} career pages.")
-    print(f"💤 Web crawler has finished.")
+    logging.info(f"Keywords were found in job listings at {kws_found_count} out of {len(careerPages)} career pages.")
 
-def close_dialogs(driver):
+
```

Comment:
Please add back in the final line. It can be a log:
logging.info(f"💤 Web crawler has finished.")